x-pack/test/alerting_api_integration/security_and_spaces/group2/tests/actions/connector_types/bedrock.ts

```diff
@@ -0,0 +1,444 @@
+            expect(resp.body).to.eql({
+              statusCode: 400,
+              error: 'Bad Request',
+              message:
+                'error validating action type secrets: [accessKey]: expected value of type [string] but got [undefined]',
+            });
+          });
+      });
+      it('should return 400 Bad Request when creating the connector without secret secret', async () => {
+        await supertest
+          .post('/api/actions/connector')
+          .set('kbn-xsrf', 'foo')
+          .send({
+            name,
+            connector_type_id: connectorTypeId,
+            config,
+            secrets: {
+              accessKey: 'accessKey',
+            },
+          })
+          .expect(400)
+          .then((resp: any) => {
+            expect(resp.body).to.eql({
+              statusCode: 400,
+              error: 'Bad Request',
+              message:
+                'error validating action type secrets: [secret]: expected value of type [string] but got [undefined]',
+            });
+          });
+      });
+    });
+
+    describe('executor', () => {
+      describe('validation', () => {
+        const simulator = new BedrockSimulator({
+          proxy: {
+            config: configService.get('kbnTestServer.serverArgs'),
+          },
+        });
+        let bedrockActionId: string;
+
+        before(async () => {
+          const apiUrl = await simulator.start();
+          bedrockActionId = await createConnector(apiUrl);
+        });
+
+        after(() => {
+          simulator.close();
+        });
+
+        it('should fail when the params is empty', async () => {
+          const { body } = await supertest
+            .post(`/api/actions/connector/${bedrockActionId}/_execute`)
+            .set('kbn-xsrf', 'foo')
+            .send({
+              params: {},
+            });
+          expect(200);
+
+          expect(body).to.eql({
+            status: 'error',
+            connector_id: bedrockActionId,
+            message:
+              'error validating action params: [subAction]: expected value of type [string] but got [undefined]',
+            retry: false,
+          });
+        });
+
+        it('should fail when the subAction is invalid', async () => {
+          const { body } = await supertest
+            .post(`/api/actions/connector/${bedrockActionId}/_execute`)
+            .set('kbn-xsrf', 'foo')
+            .send({
+              params: { subAction: 'invalidAction' },
+            })
+            .expect(200);
+
+          expect(body).to.eql({
+            connector_id: bedrockActionId,
+            status: 'error',
+            retry: true,
+            message: 'an error occurred while running the action',
+            service_message: `Sub action "invalidAction" is not registered. Connector id: ${bedrockActionId}. Connector name: AWS Bedrock. Connector type: .bedrock`,
+          });
+        });
+      });
+
+      describe('execution', () => {
```

Comment:
maybe a test to ensure model overrides are respected when executing or invoking?